docs/tutorials/excel-tutorial-create-custom-functions.md

```diff
@@ -3,8 +3,293 @@ title: Excel custom functions tutorial
+
+In your Excel workbook, try out the `ADD` custom function by completing the following steps in Excel:
+
+1. Within a cell, type **=CONTOSO**. Notice that the autocomplete menu shows the list of all functions in the `CONTOSO` namespace.
+
+2. Run the `CONTOSO.ADD` function, with numbers `10` and `200` as input parameters, by typing the value `=CONTOSO.ADD(10,200)` in the cell and pressing enter.
+
+The `ADD` custom function computes the sum of the two numbers that you specify as input parameters. Typing `=CONTOSO.ADD(10,200)` should produce the result **210** in the cell after you press enter.
+
+## Create a custom function that requests data from the web
+
+What if you needed a function that could request the price of a stock from an API and display the result in the cell of a worksheet? Custom functions are designed so that you can easily request data from the web asynchronously.
+
+Complete the following steps to create a custom function named `stockPrice` that accepts a stock ticker symbol (e.g., **MSFT**) and returns the price of that stock. This custom function uses the IEX Trading API, which is free and does not require authentication.
+
+1. In the **stock-ticker** project that the Yeoman generator created, find the file **src/customfunctions.js** and open it in your code editor.
+
+2. In **customfunctions.js**, locate the `increment` function and add the following code immediately after that function.
+
+    ```js
+    function stockPrice(ticker) {
+        var url = "https://api.iextrading.com/1.0/stock/" + ticker + "/price";
+        return fetch(url)
+            .then(function(response) {
+                return response.text();
+            })
+            .then(function(text) {
+                return parseFloat(text);
+            });
+
+        // Note: in case of an error, the returned rejected Promise
+        //    will be bubbled up to Excel to indicate an error.
+    }
+
+3. In **customfunctions.js**, locate the line`CustomFunctionMappings.INCREMENT = increment;`, add the following line of code immediately after that line, and save the file.
+
+    ```js
+    CustomFunctionMappings.STOCKPRICE = stockPrice;
+    ```
+
+4. Before Excel can make this new function available, you must specify metadata to describe the function to Excel. Open the **config/customfunctions.json** file. Add the following JSON object to the 'functions' array and save the file.
+
+    This JSON describes the `stockPrice` function.
+
+    ```JSON
+    {
+        "id": "STOCKPRICE",
+        "name": "STOCKPRICE",
+        "description": "Fetches current stock price",
+        "helpUrl": "http://www.contoso.com/help",
+        "result": {
+            "type": "number",
+            "dimensionality": "scalar"
+        },  
+        "parameters": [
+            {
+                "name": "ticker",
+                "description": "stock ticker name",
```

Comment:
Isn't this parameter technically the stock ticker symbol?

Reply:
This PR is mainly refactoring the *file structure* of the existing tutorial; I'll defer this feedback to @mscharlock for future consideration.

Reply:
Yep; will update to "stock symbol"